### ISSUE TITLE
Well known extensions should likely not camel-case match in Nav to.

### ISSUE THREAD

**Issue by CyrusNajmabadi** [dotnet/roslyn#19602]
If i write a filename like "MockSymWriter.cs" in NavigateTo, we likely should not be trying to match "cs" against symbols in the MockSymWriter type.

**Comment by sharwell** [comment]
Is this causing a specific problem? My concern here is it sounds like you may be trying to fix a *selection* bug by changing the way *filtering* works.

**Comment by CyrusNajmabadi** [comment]
> Is this causing a specific problem

Yes.  It's causing poor results to show up in the UI.  I would prefer to not present them as they just end up being confusing, and it means it's something the user has to sidestep while the proper results come in.

**Comment by sharwell** [comment]
What if the name of the member is `MockSymWriter.CSharpDefault`? If you want filenames, type `f` before your search and it will only show files in the result.

**Comment by CyrusNajmabadi** [comment]
i'm ok with not showing the item in that case.  i think a general rule of something CamelHumped with a dot, with a two letter lowercased string that is cs or vb is fine to filter out.
